saagieapi/repositories/repositories.py

```diff
@@ -0,0 +1,301 @@
+import logging
+from pathlib import Path
+from typing import Dict, Optional
+
+from gql import gql
+
+from .gql_queries import *
+
+
+class Repositories:
+    def __init__(self, saagie_api):
+        self.saagie_api = saagie_api
+
+    def list(self, pprint_result: Optional[bool] = None) -> Dict:
+        """
+        Get information for all repositories
+        NB: You can only get repositories information if you have the right to
+        access the technology catalog
+
+        Parameters
+        ----------
+        pprint_result : bool, optional
+            Whether to pretty print the result of the query, default to
+            saagie_api.pprint_global
+
+        Returns
+        -------
+        dict
+            Dict of repositories
+        """
+        query = gql(GQL_LIST_REPOSITORIES)
+        return self.saagie_api.client_gateway.execute(query, pprint_result=pprint_result)
+
+    def list_minimal(self) -> Dict:
+        """List only repository names, ids and last reversible IDs.
+        NB: You can only list repository if you have at least the viewer role on the
+        catalog
+
+        Returns
+        -------
+        dict
+            Dict of repository ids, names and last reversible IDs
+        """
+
+        return self.saagie_api.client_gateway.execute(query=gql(GQL_LIST_REPOSITORIES_MINIMAL))
```

Comment:
I think you should use fragment in your graphql request to merge these 2 functions.
You can show how it is done for the apps

Reply:
done